### PR TITLE
New package: helvum-0.3.4

### LINKED ISSUE #35582: [Package request] Helvum
Helvum is a convenient and easy to use patchbay for pipewire. It is available at: https://gitlab.freedesktop.org/pipewire/helvum

Have a lovely day

### PR DESCRIPTION
**Pull request by classabbyamp**
closes #35582
closes #34560

#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by dkwo** [comment]
Does `gtk4-update-icon-cache` help?

**Comment by dkwo** [comment]
Or maybe these two need to go to `host`: appstream-glib-devel gtk-update-icon-cache?

**Comment by classabbyamp** [comment]
still some issues on cross, but need to work on other things for now

i686 just needs [this patch](https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/merge_requests/108) for the pipewire crate

fwiw [alpine just gave up](https://git.alpinelinux.org/aports/tree/testing/helvum/APKBUILD#n10) on i686 and aarch32

**Comment by dkwo** [comment]
Does it make any difference if you add pipewire-devel to host as well?

**Comment by dkwo** [comment]
Just for future record: they both fail with
```
error: failed to run custom build command for `libspa-sys v0.4.1 (/builddir/helvum-0.3.4/libspa-sys)`

Caused by:
  process didn't exit successfully: `/builddir/helvum-0.3.4/build/src/release/build/libspa-sys-8591670d6ba9e6b2/build-script-build` (exit status: 101)
--- stderr
  /usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
  /usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', libspa-sys/build.rs:37:39
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

**Comment by 0x5c** [comment]
I did some research, and it seems that libspa-sys is inherently nocross, for now.

I uses bindgen to generate the bindings, but does no effort whatsoever to have bindgen use the target toolchain, resulting in the bindings being generated for the host.

This seems to work fine for musl, but glibc is a fragile thing that needs special handling. In this case stubs.h contains this
```h
/* This file is automatically generated.
   This file selects the right generated file of `__stub_FUNCTION' macros
   based on the architecture being compiled for.  */


#if !defined __x86_64__
# include <gnu/stubs-32.h>
#endif
#if defined __x86_64__ && defined __LP64__
# include <gnu/stubs-64.h>
#endif
#if defined __x86_64__ && defined __ILP32__
# include <gnu/stubs-x32.h>
#endif
```
... which assumes 32bit when 64bit isn't defined

It seems that the musl bindings on cross might be broken too, just not in ways that prevents compilation. I would say it's nocross everywhere unless someone can test that musl cross produces working binaries.


Outside of that, meson is only being used a a very very very thin wrapper around cargo, and to update gtk's icon cache.
The first does no useful work, and the second should be done when installing the package. Switching the build style to cargo seems to work fine (apart from the existing cross issue), and removes the need for meson and the icon stuff in hostmakedeps.

**Comment by classabbyamp** [comment]
If that's the case, I'd support nocross for now, and I think nixing meson will work, but may require future care in the case of added dependencies on updates

**Comment by 0x5c** [comment]
Reported the crossbuild issue to upstream https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/issues/40

**Comment by classabbyamp** [comment]
@paper42, this should be ready for review

**Comment by classabbyamp** [comment]
seem to have found a fix for the bindgen/cross issue. tested briefly on an rpi3 (aarch64-glibc)

File: srcpkgs/helvum/template [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Template file for 'helvum'
+pkgname=helvum
+version=0.3.4
+revision=1
+build_style=cargo
+build_helper=rust
+hostmakedepends="pkg-config clang libclang"
+makedepends="gtk4-devel pipewire-devel glib-devel"
+depends="gtk4 pipewire"
```
Comment (paper42) [inline srcpkgs/helvum/template:9]:
```suggestion
depends="pipewire"
```
gtk4 is automatically detected

File: srcpkgs/helvum/template [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+short_desc="GTK patchbay for pipewire"
+maintainer="classabbyamp <dev@kb6.ee>"
+license="GPL-3.0-only"
+homepage="https://gitlab.freedesktop.org/pipewire/helvum"
+distfiles="https://gitlab.freedesktop.org/pipewire/helvum/-/archive/${version}/helvum-${version}.tar.gz
+ https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/archive/v0.4.1/pipewire-rs-v0.4.1.tar.gz"
```
Comment (paper42) [inline srcpkgs/helvum/template:15]: could you create a variable _pipewire_rs_version, put it next to $version and use it here?

Comment (classabbyamp) [reply srcpkgs/helvum/template]: done.

File: srcpkgs/helvum/template [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	vinstall data/$_app_id.desktop.in 644 usr/share/applications/ $_app_id.desktop
+
+	vsed -i data/$_app_id.metainfo.xml.in -e "s/@app-id@/$_app_id/"
+	vinstall data/$_app_id.metainfo.xml.in 644 usr/share/metainfo/ $_app_id.metainfo.xml
+}
+
+do_clean() {
+	# clean up pipewire-rs
+	rm -rf ${XBPS_BUILDDIR}/pipewire-rs-*
+}
```
Comment (paper42) [inline srcpkgs/helvum/template:47]: We can use create_wrksrc instead.

Comment (classabbyamp) [reply srcpkgs/helvum/template]: this works but it causes the patch to need `/helvum-0.3.4/` added to the path. see the latest pushed version.

**Pull request merged**